src/main/java/com/cleanroommc/groovyscript/compat/mods/alchemistry/Atomizer.java

```diff
@@ -0,0 +1,131 @@
+package com.cleanroommc.groovyscript.compat.mods.alchemistry;
+
+import al132.alchemistry.recipes.AtomizerRecipe;
+import al132.alchemistry.recipes.LiquifierRecipe;
+import al132.alchemistry.recipes.ModRecipes;
+import com.cleanroommc.groovyscript.api.GroovyLog;
+import com.cleanroommc.groovyscript.api.IIngredient;
+import com.cleanroommc.groovyscript.api.documentation.annotations.*;
+import com.cleanroommc.groovyscript.compat.mods.ModSupport;
+import com.cleanroommc.groovyscript.helper.SimpleObjectStream;
+import com.cleanroommc.groovyscript.helper.ingredient.IngredientHelper;
+import com.cleanroommc.groovyscript.helper.recipe.AbstractRecipeBuilder;
+import com.cleanroommc.groovyscript.registry.VirtualizedRegistry;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.fluids.FluidStack;
+import org.jetbrains.annotations.Nullable;
+
+@RegistryDescription
+public class Atomizer extends VirtualizedRegistry<AtomizerRecipe> {
+
+    @Override
+    public void onReload() {
+        removeScripted().forEach(recipe -> ModRecipes.INSTANCE.getAtomizerRecipes().removeIf(r -> r == recipe));
+        ModRecipes.INSTANCE.getAtomizerRecipes().addAll(restoreFromBackup());
+    }
+
+    @RecipeBuilderDescription(example = {
+            @Example(".fluidInput(fluid('water') * 125).output(item('minecraft:clay'))"),
+            @Example(".fluidInput(fluid('lava') * 500).output(item('minecraft:gold_ingot')).reversible()")
+    })
+    public RecipeBuilder recipeBuilder() {
+        return new RecipeBuilder();
+    }
+
+    public AtomizerRecipe add(AtomizerRecipe recipe) {
+        if (recipe != null) {
+            addScripted(recipe);
+            ModRecipes.INSTANCE.getAtomizerRecipes().add(recipe);
+        }
+        return recipe;
+    }
+
+    public boolean remove(AtomizerRecipe recipe) {
+        if (ModRecipes.INSTANCE.getAtomizerRecipes().removeIf(r -> r == recipe)) {
+            addBackup(recipe);
+            return true;
+        }
+        return false;
+    }
+
+    @MethodDescription(description = "groovyscript.wiki.removeByOutput", example = @Example(value = "item('alchemistry:compound:7')", commented = true))
+    public boolean removeByOutput(ItemStack output) {
```

Comment:
Rather use `IIngredient` and `.test()`. This applies to other machines for input and output as well.

Reply:
alright, changed them. there are a few parts of older compat that will also need this change. uh, most really.